physics/module_bl_mynn.F90

```diff
@@ -558,37 +544,32 @@ SUBROUTINE mynn_bl_driver(            &
 !                        =   3;  Level 3
     
     REAL, INTENT(in) :: delt
-!WRF
-!    REAL, INTENT(in) :: dx
-!END WRF
-!FV3
     REAL, DIMENSION(IMS:IME), INTENT(in) :: dx
-!END FV3
-    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(in) :: dz,&
+    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(in) :: dz,      &
          &u,v,w,th,sqv3D,p,exner,rho,T3D
-    REAL, DIMENSION(IMS:IME,KMS:KME), OPTIONAL, INTENT(in)::&
+    REAL, DIMENSION(IMS:IME,KMS:KME), OPTIONAL, INTENT(in):: &
          &sqc3D,sqi3D,qni,qnc,qnwfa,qnifa
     REAL, DIMENSION(IMS:IME,KMS:KME), OPTIONAL, INTENT(in):: ozone
-    REAL, DIMENSION(IMS:IME), INTENT(in) :: xland,ust,&
+    REAL, DIMENSION(IMS:IME), INTENT(in) :: xland,ust,       &
          &ch,ts,qsfc,ps,hfx,qfx,wspd,uoce,voce,vdfg,znt
 
-    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(inout) :: &
-         &Qke,Tsq,Qsq,Cov,qke_adv     !ACF for QKE advection
+    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(inout) ::       &
+         &Qke,Tsq,Qsq,Cov,qke_adv
 
-    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(inout) :: &
-         &RUBLTEN,RVBLTEN,RTHBLTEN,RQVBLTEN,RQCBLTEN,&
-         &RQIBLTEN,RQNIBLTEN,RQNCBLTEN, &
+    REAL, DIMENSION(IMS:IME,KMS:KME), INTENT(inout) ::       &
+         &RUBLTEN,RVBLTEN,RTHBLTEN,RQVBLTEN,RQCBLTEN,        &
+         &RQIBLTEN,RQNIBLTEN,RQNCBLTEN,                      &
          &RQNWFABLTEN,RQNIFABLTEN
```

Comment:
Dummy argument arrays should be assumed shape arrays (`dimension(:,:)`) so the intel compiler bounds checking will detect out-of-bounds accesses. This problem is already throughout the code, so there's no need to fix it in this commit.

Reply:
No... That doesn't work in WRF, so it screws up the "universality", so I will not conform to that rule. I will conform to that rule only in CCPP-specific modules, like the wrapper.

Reply:
For it to work with WRF, you would use `dimension(IMS:,KMS:)`, which is an assumed shape array with a specified start index. You'll still get the benefit of bounds checking.

Reply:
I'll try it. Does that form conform to the CCPP standards too?

Reply:
As far as I know, yes. I wouldn't include that in this PR though.